### PR TITLE
TE5-T447 Jetpack Navigation Component Added

### PR DESCRIPTION
**Pull request by jay-a-shah**

### CODE REVIEW

File: app/src/main/java/com/example/demoappkotlin/listView_ViewPager/Fragments/GalleryFragment.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import android.widget.Button
+import androidx.navigation.Navigation
+import androidx.navigation.findNavController
+import com.example.demoappkotlin.R
+import com.example.demoappkotlin.activityAndFragments.JetPackNavigationActivity
+
+class GalleryFragment : Fragment() {
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/listView_ViewPager/Fragments/GalleryFragment.kt]: code formatting

**Review by purvesh-d — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/demoappkotlin/MainActivity.kt [hunk 2]
```diff
@@ -30,7 +31,11 @@ class MainActivity : AppCompatActivity() {
         //Android Ui Button with Normal toast
         val btnAndroidUI = findViewById<Button>(R.id.btnAndroidUi)
         btnAndroidUI.setOnClickListener {
-            val normalToast = Toast.makeText(applicationContext, getString(R.string.toast_normal), Toast.LENGTH_SHORT)
+            val normalToast = Toast.makeText(
```
Comment (Purvesh-Dodiya) [inline app/src/main/java/com/example/demoappkotlin/MainActivity.kt]: Why you made this multiple lines and removed single line?

File: app/src/main/java/com/example/demoappkotlin/listView_ViewPager/Fragments/GalleryFragment.kt [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import androidx.navigation.findNavController
+import com.example.demoappkotlin.R
+import com.example.demoappkotlin.activityAndFragments.JetPackNavigationActivity
+
+class GalleryFragment : Fragment() {
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        // Inflate the layout for this fragment
+        val view = inflater.inflate(R.layout.fragment_gallery, container, false)
+        view.findViewById<Button>(R.id.btnNextFrag).setOnClickListener(Navigation.createNavigateOnClickListener(R.id.action_galleryFragment_to_addMessageFragment))
```
Comment (Purvesh-Dodiya) [inline app/src/main/java/com/example/demoappkotlin/listView_ViewPager/Fragments/GalleryFragment.kt]: You can put this onClick inside `onViewCreated {}`

File: app/src/main/res/layout/fragment_add_message_with_safe_args.xml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        android:layout_margin="15dp"
+        android:text="@string/send_feedback_btn"
+        android:textAppearance="@style/TextAppearance.AppCompat.Display1"
+        android:textColor="@color/text_color_hint"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/textview_Recent_Fragment" />
+
+    <androidx.appcompat.widget.AppCompatTextView
+        android:id="@+id/textview_Recent_Fragment"
```
Comment (Purvesh-Dodiya) [inline app/src/main/res/layout/fragment_add_message_with_safe_args.xml]: undescore??

File: app/src/main/res/navigation/nav_graph.xml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/navigation"
+    app:startDestination="@id/galleryFragmentJetPackNavCompose">
+
+    <fragment
+        android:id="@+id/galleryFragmentJetPackNavCompose"
+        android:name="com.example.demoappkotlin.listView_ViewPager.Fragments.GalleryFragment"
+        android:label="@string/label_gallery_fragment"
+        tools:layout="@layout/fragment_gallery">
+        <action
```
Comment (Purvesh-Dodiya) [inline app/src/main/res/navigation/nav_graph.xml:13]: apply code formatting

**Review by Purvesh-Dodiya — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/demoappkotlin/listView_ViewPager/Fragments/GalleryFragment.kt [hunk 6]
```diff
@@ -0,0 +1,20 @@
+package com.example.demoappkotlin.listView_ViewPager.Fragments
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/listView_ViewPager/Fragments/GalleryFragment.kt:5]: unused import are there

**Review by purvesh-d — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/demoappkotlin/MainActivity.kt [hunk 7]
```diff
@@ -64,6 +67,17 @@ class MainActivity : AppCompatActivity() {
...
+        btnJetPackNavComponent.setOnClickListener {
+            val intent = Intent(applicationContext, JetPackNavigationActivity::class.java)
+            startActivity(intent)
+        }
+        (intent.getParcelableExtra<Parcelable>(Intent.EXTRA_STREAM) as? Uri)?.let {
+            val intent = Intent(applicationContext,ImplicitIntent::class.java)
+            intent.putExtra("Key",it)
+            startActivity(intent)
+        }
+
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/MainActivity.kt]: extra empty lines

File: app/src/main/java/com/example/demoappkotlin/activityAndFragments/ImplicitIntent.kt [hunk 8]
```diff
@@ -25,12 +26,14 @@ class ImplicitIntent : AppCompatActivity() {
                 resultLauncher.launch((it))
             }
         }
-    }
+        var imageUri = intent.getParcelableExtra<Parcelable>("Key")
+        ivphoto.setImageURI(imageUri as Uri?)
+        }
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/activityAndFragments/ImplicitIntent.kt]: code formatting

**Review by purvesh-d — CHANGES_REQUESTED** [review]

**Review by purvesh-d — APPROVED** [review]

**Pull request merged**